src/status_im/data_store/messages.cljs

```diff
@@ -95,14 +95,17 @@
 
 (defn save-message-tx
   "Returns tx function for saving message"
-  [{:keys [message-id from] :as message}]
-  (fn [realm]
-    (core/create realm
-                 :message
-                 (prepare-message (merge default-values
-                                         message
-                                         {:from (or from "anonymous")}))
-                 true)))
+  ([message]
+   (save-message-tx message nil))
+  ([{:keys [message-id from] :as message} on-success]
```

Comment:
here we could try something like:

```
{:success-event [:message-confirmed raw-message] ; can be passed as a parameter 
  :transaction (fn [realm] .....)}
```
and theoretically it might just be the only change we need